sdk/identity/azure-identity/azure/identity/_credentials/azure_arc.py

```diff
@@ -0,0 +1,172 @@
+        if client_args:
+            self._client = ManagedIdentityClient(**client_args)
+        else:
+            self._client = None
+
+    def get_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        if not self._client:
+            raise CredentialUnavailableError(
+                message="Azure Arc managed identity configuration not found in environment"
+            )
+        return super(AzureArcCredential, self).get_token(*scopes, **kwargs)
+
+    def _acquire_token_silently(self, *scopes):
+        # type: (*str) -> Optional[AccessToken]
+        return self._client.get_cached_token(*scopes)
+
+    def _request_token(self, *scopes, **kwargs):
+        # type: (*str, **Any) -> AccessToken
+        return self._client.request_token(*scopes, **kwargs)
+
+
+def _get_policies(credential, config, **kwargs):
+    # type: (AzureArcCredential, Configuration, **Any) -> List[PolicyType]
+    return [
+        HeadersPolicy(**kwargs),
+        UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs),
+        config.proxy_policy,
+        config.retry_policy,
+        NetworkTraceLoggingPolicy(**kwargs),
+        DistributedTracingPolicy(**kwargs),
+        HttpLoggingPolicy(**kwargs),
+        ArcChallengeAuthPolicy(credential),
+    ]
+
+
+def _get_client_args(credential, **kwargs):
+    # type: (AzureArcCredential, **Any) -> Optional[dict]
+    identity_config = kwargs.pop("_identity_config", None) or {}
+
+    url = os.environ.get(EnvironmentVariables.IDENTITY_ENDPOINT)
+    if not url:
+        # Azure Arc managed identity isn't available in this environment
+        return None
+
+    config = _get_configuration()
+
+    return dict(
+        kwargs,
+        _identity_config=identity_config,
+        policies=_get_policies(credential, config),
+        request_factory=functools.partial(_get_request, url),
+    )
+
+
+def _get_request(url, scope, identity_config):
+    # type: (str, str, dict) -> HttpRequest
+    request = HttpRequest("GET", url)
+    request.format_parameters(dict({"api-version": "2019-08-15", "resource": scope}, **identity_config))
+    return request
+
+
+def _update_headers(request, secret_key=None):
+    # type: (PipelineRequest, str) -> PipelineRequest
+    request.http_request.headers = {"Metadata": "true"}
+    if secret_key:
+        request.http_request.headers["Authorization"] = "Basic {}".format(secret_key)
+
+    return request
+
+
+def _get_secret_key(response):
+    # type: (PipelineResponse) -> str
+    # expecting header containing path to secret key file
+    header = response.http_response.headers.get("Www-Authenticate")
+    if header is None:
+        raise CredentialUnavailableError(message="Did not receive a value from Www-Authenticate header")
+
+    secret_key = None
+
+    # expecting header with structure like 'Basic realm=<file path>'
+    key_file = header.split("=")[1]
+    with open(key_file, "r") as file:
+        try:
+            secret_key = file.read()
```

Comment:
Yeah, returning here makes more sense. I was going to, but didn't because of the potential error on the read operation -- but that doesn't actually make any sense in retrospect